### PR TITLE
Refactoring: rename home meter to smart meter

### PR DESCRIPTION
**Pull request by fievelk**
This refactoring was asked by the stakeholders to homogenize our terminology to the energy market's.

## Related PRs

- https://github.com/gridsingularity/d3a-interface/pull/249
- https://github.com/gridsingularity/d3a-web/pull/1433

### CODE REVIEW

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

File: src/d3a/setup/strategy_tests/smart_meter_setup.py [hunk 1]
```diff
@@ -37,9 +37,9 @@ def get_setup(config):
                 grid_fee_percentage=0, grid_fee_constant=0,
             ),
             Area("House 2", [
-                Area("H2 Home Meter", strategy=HomeMeterStrategy(
+                Area("H2 Smart Meter", strategy=SmartMeterStrategy(
```
Comment (faizan2590) [inline src/d3a/setup/strategy_tests/smart_meter_setup.py:40]: An additional nesting of area for House2 could be avoided here as House 2 anyway have lumped energy reading of house via `H2 Smart Meter`

**Review by faizan2590 — DISMISSED** [review]
One minor enhancement comments, other than that LGTM.

File: src/d3a/setup/strategy_tests/smart_meter_setup.py (hunk 1 above)
Comment (faizan2590) [reply src/d3a/setup/strategy_tests/smart_meter_setup.py:40]: Any response to this comment?

**Review by faizan2590 — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**